packages/find-root/src/index.ts

```diff
@@ -19,28 +19,54 @@ import {
  * monorepo implementations first, with tools based on custom file schemas
  * checked last.
  */
-const defaultOrder: Tool[] = [YarnTool, PnpmTool, LernaTool, RushTool];
+const DEFAULT_TOOLS: Tool[] = [
+  YarnTool,
+  PnpmTool,
+  LernaTool,
+  RushTool,
+  RootTool,
+];
 
 const isNoEntryError = (err: unknown): boolean =>
   !!err && typeof err === "object" && "code" in err && err.code === "ENOENT";
 
 export class NoPkgJsonFound extends Error {
+  directory: string;
+  constructor(directory: string) {
+    super(`No package.json could be found upwards from directory ${directory}`);
+    this.directory = directory;
+  }
+}
+
+export class NoMatchingMonorepoFound extends Error {
   directory: string;
   constructor(directory: string) {
     super(
-      `No package.json could be found upwards from the directory ${directory}`
+      `No monorepo matching the list of supported monorepos could be found upwards from directory ${directory}`
     );
     this.directory = directory;
   }
 }
 
-export async function findRoot(cwd: string): Promise<MonorepoRoot> {
+/**
+ * Given a starting folder, search that folder and its parents until a supported monorepo
+ * is found, and return a `MonorepoRoot` object with the discovered directory and a
+ * corresponding monorepo `Tool` object.
+ *
+ * By default, all predefined `Tool` implementations are included in the search -- the
+ * caller can provide a list of desired tools to restrict the types of monorepos discovered,
+ * or to provide a custom tool implementation.
+ */
+export async function findRoot(
+  cwd: string,
+  tools: Tool[] = DEFAULT_TOOLS
```

Comment:
What do you think about making this argument an object? I'm worried that using a position argument here might become cumbersome in the future if we ever end up wanting to add a new parameter

Reply:
Kick off an options interface? Makes sense to me, I'll put that in this weekend.